app/src/main/java/com/hwang/xsighting/CreateSighting.java

```diff
@@ -0,0 +1,200 @@
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+import com.google.firebase.firestore.DocumentReference;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.GeoPoint;
+import com.hwang.xsighting.models.Sighting;
+
+import java.io.IOException;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+public class CreateSighting extends AppCompatActivity {
+
+    private final String TAG = "Create.Sighting";
+    private FirebaseFirestore db = FirebaseFirestore.getInstance();
+    private FusedLocationProviderClient fusedLocationClient;
+    private final int MY_PERMISSIONS_REQUEST_LOCATIONS = 1122;
+    private String lastLocation;
+    private GeoPoint geoPointLocation;
+    private static FirebaseUser user;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_create_sighting);
+
+        // Get Location
+        fusedLocationClient = LocationServices.getFusedLocationProviderClient(this);
+        getLocation();
+        // Set the username
+        user = FirebaseAuth.getInstance().getCurrentUser();
+    }
+
+
+    public void submitSighting(View view) {
+        Date dateData = new Date();
+        Timestamp timestamp = new Timestamp(dateData);
+        EditText descriptionEditText = findViewById(R.id.report_description);
+        String description = descriptionEditText.getText().toString();
+        Sighting sighting = new Sighting(user.getUid(), user.getDisplayName(), timestamp, lastLocation, geoPointLocation, "placeholder", description );
+//        Map<String, Object> sighting = new HashMap<>();
+//        Date timestamp = new Date();
+//        EditText descriptionEditText = findViewById(R.id.report_description);
+//        String description = descriptionEditText.getText().toString();
+
+
+//        sighting.put("authorUsername", user.getDisplayName());
+//        sighting.put("createdTime", timestamp);
+//        sighting.put("description", description);
+//        sighting.put("image_url", "placeholder");
+//        sighting.put("locationData", geoPointLocation);
+//        sighting.put("authorId", user.getUid());
+//        sighting.put("city", lastLocation);
+
+
+        // Add a new document with a generated ID
+        db.collection("sighting")
+                .add(sighting)
+                .addOnSuccessListener(new OnSuccessListener<DocumentReference>() {
+                    @Override
+                    public void onSuccess(DocumentReference documentReference) {
+                        Log.d(TAG, "DocumentSnapshot added with ID: " + documentReference.getId());
+                        // Toast success message

```

Comment:
Where did this code come from? Citation needed. (You wouldn't declare this as a CharSequence.)